### PR TITLE
Blazor ebook - AppStartup chapter

### PR DESCRIPTION
**Pull request by csharpfritz**
## Summary

Introducing the App-Startup chapter for the Blazor for Web Forms developer architecture book

### CODE REVIEW

**Comment by BillWagner** [comment]
Changing the milestone because this is still a draft PR.

**Review by danroth27 — CHANGES_REQUESTED** [review]
Looks good! Just a few comments.

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (csharpfritz) [reply docs/architecture/blazor-for-web-forms-developers/app-startup.md:25]: If this ebook is _only_ targeting migrating from Web Forms to Blazor Server, I think we should just keep it clear that this is a Blazor Server configuration.

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (csharpfritz) [reply docs/architecture/blazor-for-web-forms-developers/app-startup.md]: Added middleware reference and link

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (csharpfritz) [reply docs/architecture/blazor-for-web-forms-developers/app-startup.md]: Added sentence to reference the client-side routing handling

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (danroth27) [reply docs/architecture/blazor-for-web-forms-developers/app-startup.md]: Are we scoping this book to just Blazor Server? I don't see any reason why we would do that with Blazor WebAssembly shipping in just a few weeks.

**Comment by BillWagner** [comment]
@danroth27 I believe all your comments have been addressed. Can you take a fresh look?

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (danroth27) [inline docs/architecture/blazor-for-web-forms-developers/app-startup.md:88]:
```suggestion
The final statement in this method defines the endpoints that ASP.NET Core is listening on.  These are the web accessible locations that you can access on the web server and receive some content handled by .NET and returned to you.  The first entry, `MapBlazorHub` configures a SignalR hub for use in providing the real-time and persistent connection to the server where the state and rendering of Blazor components is handled.  The `MapFallbackToPage` method call indicates the web-accessible location of the page that starts the Blazor application and also configures the application to handle deep-linking requests from the client-side.  You will see this feature at work if you open a browser and navigate directly to Blazor handled route in your application, such as `/counter` in the default project template. The request gets handled by the *_Host.cshtml* fallback page, which then runs the Blazor router and renders the counter page.
```

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (danroth27) [inline docs/architecture/blazor-for-web-forms-developers/app-startup.md:92]:
```suggestion
Technologies for bundling assets like CSS stylesheets and JavaScript files have evolved significantly, with other technologies providing quickly evolving tools and techniques for managing these resources.  To this end, there are several techniques recommended for managing these resources:
```

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (danroth27) [inline docs/architecture/blazor-for-web-forms-developers/app-startup.md:105]:
```suggestion
More details about both strategies to manage your CSS and JavaScript files are available in the [Bundle and minify static assets in ASP.NET Core](https://docs.microsoft.com/aspnet/core/client-side/bundling-and-minification) documentation.
```

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (danroth27) [inline docs/architecture/blazor-for-web-forms-developers/app-startup.md:95]: We no longer recommend this tool in our latest guidance.

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (danroth27) [inline docs/architecture/blazor-for-web-forms-developers/app-startup.md:25]: Now that Blazor WebAssembly has shipped, I think we need a Blazor WebAssembly section as well. Blazor WebAssembly apps define startup logic in Program.Main and don't use the same `Startup` pattern used by ASP.NET Core. Technically both Blazor Server and Blazor WebAssembly define a `Program.Main` that sets up the host for the app. Blazor Server apps use the normal ASP.NET Core host, while Blazor WebAssembly apps have a WebAssembly specific host.

**Review by danroth27 — CHANGES_REQUESTED** [review]
Looks good! Just a few comments.

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (csharpfritz) [reply docs/architecture/blazor-for-web-forms-developers/app-startup.md:25]: This feels like scope creep at this point, and I suggest we recommend adding a chapter or two after the first version ships

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (csharpfritz) [reply docs/architecture/blazor-for-web-forms-developers/app-startup.md]: Removed reference to the bundle tool and reworded the prior sentence

**Comment by csharpfritz** [comment]
Tagging @danroth27 to review comments after merge

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (danroth27) [reply docs/architecture/blazor-for-web-forms-developers/app-startup.md:25]: We already talk about Blazor WebAssembly in the earlier chapters in the book, but we can unblock this PR and add the content later. Filed https://github.com/dotnet/docs/issues/19842 to track.

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (danroth27) [inline docs/architecture/blazor-for-web-forms-developers/app-startup.md:78]:
```suggestion
The `ConfigureServices` method introduced in ASP.NET Core allows for the various ASP.NET Core framework services to be configured for the framework's built-in dependency injection container.  The various `services.Add*` methods add services that enable features such as authentication, razor pages, MVC controller routing, SignalR, and Blazor Server interactions among many others.  This method was not needed in web forms, as the parsing and handling of the ASPX, ASCX, ASHX, and ASMX files was defined by referencing ASP.NET in the web.config configuration file.  More information about dependency injection in ASP.NET Core is available in the [online documentation](https://docs.microsoft.com/aspnet/core/fundamentals/dependency-injection).
```

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (danroth27) [inline docs/architecture/blazor-for-web-forms-developers/app-startup.md:70]: Missing indentation:

```suggestion
      endpoints.MapBlazorHub();
      endpoints.MapFallbackToPage("/_Host");
```

**Review by danroth27 — APPROVED** [review]
:shipit:

**Review by nishanil — APPROVED** [review]
Thanks @csharpfritz and @danroth27

File: docs/architecture/blazor-for-web-forms-developers/app-startup.md
Comment (nishanil) [inline docs/architecture/blazor-for-web-forms-developers/app-startup.md:86]:
```suggestion
The next line is the first that replicates one of the configuration options from web forms: `UseRouting`.  This method adds the ASP.NET Core router to the pipeline and it can be either configured here or in the individual files that it can consider routing to.  More information about routing configuration can be found in the [Routing section](pages-routing-layouts.md).
```

**Comment by nishanil** [comment]
@csharpfritz Link to the `Routing Section` was broken. Just fixed it.

**Pull request merged**
